pyproject.toml

```diff
@@ -46,6 +47,8 @@ dependencies = [
     "prompt-toolkit;platform_system!='Emscripten'",
     "python-multipart>=0.0.7;platform_system!='Emscripten'",
     "setuptools;python_version>='3.12'",
+    "narwhals>=1.9.0,<1.10.0",
```

Comment:
Our intention with the backwards compatibility policy is that it shouldn't be necessary to set an upper bound, as you're using `narwhals.stable.v1` https://narwhals-dev.github.io/narwhals/backcompat/ 

In tomorrow's release we'd like to add `dtypes` and `typing` to `stable.v1` - with that, would you confident enough in not setting an upper-bound?
1.10.0 would likely come out next week, so this'd  be really tight 😄 

If another project were to adopt Narwhals and set - say - `1.10` as the minimum version, that'd already be enough to enter "dependency hell" 🔥

Reply:
Absolutely!

I got bit pretty hard on the 1.9 narwhals release and slightly on 1.8. But both were dtype related. 

If they move to stable, I'm very happy to remove the upper bound on the narwhals version 😊

Reply:
Thank you again for the blazing fast turn around on all of the found issues. Amazing!! 💙

Reply:
thanks! cool, will do so tomorrow morning (I have a policy to only make releases in the morning, so if there's any urgent issues I can be awake to address them 😄 it's now nearly 10 pm where I live)

we'll add py-shiny to our CI to make sure we don't break anything https://github.com/narwhals-dev/narwhals/issues/1127

Reply:
Resolved in https://github.com/posit-dev/py-shiny/pull/1717